File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteMiddle = function(head) {
    if(head.next===null){
        head=null;
        return head;
    }
    let s=head;
    let f=head.next.next;
    while(f!==null && f.next!==null){
        s=s.next;
        f=f.next.next;
    }
    s.next=s.next.next;
    return head;
};

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
    if(head.next==null){
        head=null;
        return head;
    }
    ListNode s=head;
    ListNode f=head.next.next;
    while(f!=null && f.next!=null){
        s=s.next;
        f=f.next.next;
    }
    s.next=s.next.next;
    return head;
    }
}